### PR TITLE
Сode refactoring

### PR DESCRIPTION
**Pull request by marynakernychna**
--- Add ---

> functions to access all available server endpoints
> home page
> constants

--- Change ---

> names
> alerts
> messages
> InputRules class
> file types
> header's style
> inputs' validation style
> user's cars page style

--- Delete ---

> unnecessary reducer
> menu items for which we don't have pages

--- Others ---

> fix authentication pages' style
> rewrite class components to functional components
> align code
> var to let

### CODE REVIEW

File: src/assets/scss/assets/components/_header.scss [hunk 1]
```diff
@@ -3,66 +3,70 @@
...
-@media only screen and (max-width: $max_phone_size) {
-    header{
-        display: flex;
-        gap: 0%;
-        h1{
+    @media only screen and (min-width: $max_phone_size) and (max-width: $max_tablet_size) {
+        header {
+            display: flex;
+            gap: 0%;
+            // margin-left: -8%;
```
Comment (Grasmus) [inline src/assets/scss/assets/components/_header.scss]: Did you forget to remove this comment?

File: src/components/navigation/menu.jsx [hunk 2]
```diff
@@ -1,42 +1,44 @@
...
     const logOut = () => {
         logoutUser(history);
     };
+
     if (!data) {
         return <div>Loading</div>
     }
     return (
-        <div className={`menu ${isOpen && "open"}`}>
+        <div className={`menu ${isMenuOpen && "open"}`}> {/* ??? */}
```
Comment (Grasmus) [inline src/components/navigation/menu.jsx]: What is it?

File: src/components/userCars/car/car.jsx [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+import Text from "antd/es/typography/Text";
+import bucket_icon from "../../../assets/images/cars/bucket.svg";
+import heavy_icon from "../../../assets/images/cars/heavy.svg";
+import apps_icon from "../../../assets/images/cars/apps.svg";
+import info_icon from "../../../assets/images/cars/info.svg";
+import insurance_icon from "../../../assets/images/cars/insurance.svg";
+import passport_icon from "../../../assets/images/cars/passport.svg";
+
+function Car(data) {
+    console.log(data);
```
Comment (Grasmus) [inline src/components/userCars/car/car.jsx]: Why do we need this information in console?

File: src/assets/scss/assets/components/_header.scss (hunk 1 above)
Comment (marynakernychna) [reply src/assets/scss/assets/components/_header.scss:44]: yes :(

File: src/components/navigation/menu.jsx (hunk 2 above)
Comment (marynakernychna) [reply src/components/navigation/menu.jsx:20]: the comment which I forgot to delete

File: src/components/userCars/car/car.jsx (hunk 3 above)
Comment (marynakernychna) [reply src/components/userCars/car/car.jsx:19]: I forgot to delete

File: src/components/userOffers/offer/offer.jsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+                <p className="creationDate">{moment(data.info.creationDate).format('LLL')}</p>
+                <Text strong className="offerStatus">
+
+                    {!data.info.isClosed ?
+                        <p id="opened">Opened</p> :
+                        <p id="closed">Closed</p>}
+                </Text>
+            </div>
+
+            <Tag>{data.info.creatorRoleName === "SENDER" ? "Give" : "Need"}</Tag>
```
Comment (Grasmus) [inline src/components/userOffers/offer/offer.jsx]: Move this to the constant

**Review by Grasmus — DISMISSED** [review]

File: src/assets/scss/assets/pages/_main.scss [hunk 5]
```diff
@@ -0,0 +1,81 @@
+@import '../variables/colors';
+
```
Comment (Eugene-pas) [inline src/assets/scss/assets/pages/_main.scss]: empty of line

File: src/components/newCarModal/addCarModal.jsx [hunk 6]
```diff
@@ -0,0 +1,190 @@
...
+                scrollToFirstError
+            >
+                <Form.Item name="model"
+                    label={"Model: "}
+                    rules={[
+                        InputRules.capitalLetterFirst(
+                            carsErrorMessages.CAPITAL_LETTER_FIRST
+                        ),
+                        {
+                            pattern: new RegExp("^([A-Za-z\\d ]?)*$"),
```
Comment (Eugene-pas) [inline src/components/newCarModal/addCarModal.jsx]: We have a separate file where we describe the validation rules. Let's write in one style

File: src/components/newCarModal/addCarModal.jsx [hunk 7]
```diff
@@ -0,0 +1,190 @@
...
+                    ]}
+                >
+                    <Input placeholder="Model" />
+                </Form.Item>
+
+                <Form.Item name="technicalPassport"
+                    label="Technical passport: "
+                    rules={[
+                        {
+                            pattern: new RegExp("^[a-zA-Z\\d ]*$"),
```
Comment (Eugene-pas) [inline src/components/newCarModal/addCarModal.jsx]: Similar to the previous one

File: src/components/newCarModal/addCarModal.jsx [hunk 8]
```diff
@@ -0,0 +1,190 @@
...
+                >
+                    <Input placeholder="Technical passport" />
+                </Form.Item>
+
+                <Form.Item name="registrationNumber"
+                    label={"Registration number: "}
+                    rules={[
+                        {
+                            type: "string",
+                            pattern: new RegExp("^[a-zA-Z\\d ]*$"),
```
Comment (Eugene-pas) [inline src/components/newCarModal/addCarModal.jsx]: Similar to the previous one

File: src/components/newCarModal/addCarModal.jsx [hunk 9]
```diff
@@ -0,0 +1,190 @@
...
+                    ]}
+                >
+                    <Input placeholder="Registration number" />
+                </Form.Item>
+
+                <Form.Item name="loadCapacity"
+                    label={"Load capacity: "}
+                    rules={[
+                        {
+                            pattern: new RegExp("^[^0|\\D]\\d{0,9}(\\.\\d{1,2})?$"),
```
Comment (Eugene-pas) [inline src/components/newCarModal/addCarModal.jsx]: Similar to the previous one

File: src/components/newCarModal/addCarModal.jsx [hunk 10]
```diff
@@ -0,0 +1,190 @@
...
+                        addonAfter="kg"
+                        placeholder="Load capacity"
+                    />
+                </Form.Item>
+
+                <Form.Item name="vin"
+                    label={"VIN: "}
+                    rules={[
+                        {
+                            pattern: new RegExp("^[a-zA-Z\\d]*$"),
```
Comment (Eugene-pas) [inline src/components/newCarModal/addCarModal.jsx]: Similar to the previous one

File: src/components/newCarModal/addCarModal.jsx [hunk 11]
```diff
@@ -0,0 +1,190 @@
...
+                            message:
+                                carsErrorMessages.NOT_VALID_VIN
+                        },
+                        {
+                            pattern: new RegExp("^.{17}$"),
```
Comment (Eugene-pas) [inline src/components/newCarModal/addCarModal.jsx]: Similar to the previous one

File: src/components/newCarModal/addCarModal.jsx [hunk 12]
```diff
@@ -0,0 +1,190 @@
...
+                        options={categories}
+                    >
+                    </Select>
+                </Form.Item>
+
+                <Form.Item name="color"
+                    label={"Color: "}
+                    rules={[
+                        {
+                            pattern: new RegExp("^[a-zA-Z]*$"),
```
Comment (Eugene-pas) [inline src/components/newCarModal/addCarModal.jsx]: Similar to the previous one

**Review by Eugene-pas — APPROVED** [review]

**Review by Grasmus — APPROVED** [review]

**Pull request merged**